daemon/commands.h

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#ifndef NETDATA_COMMANDS_H
+#define NETDATA_COMMANDS_H 1
+
+#ifdef _WIN32
+# define PIPENAME "\\\\?\\pipe\\netdata-cli"
+#else
+# define PIPENAME "/tmp/netdata-ipc"
+#endif
+
+#define MAX_COMMAND_LENGTH 4096
+
+typedef enum cmd {
+    CMD_HELP = 0,
+    CMD_RELOAD_HEALTH,
+    CMD_SAVE_DATABASE,
+    CMD_LOG_ROTATE,
+    CMD_EXIT,
+    CMD_FATAL,
+    CMD_TOTAL_COMMANDS
```

Comment:
Why are you putting this inside the enum and then adding BUGON later with a compile-time expression? It looks like it would be easier to just` #define (sizeof(command_info_array) / sizeof(command_info_t))`. Adding something that can go wrong, and then a `sanity_check` to stop it is not as simple as just not letting it go wrong.

Reply:
This enum indexes various data structures, it's not just a part of `command_info_array[]` (e.g. `execute_command(CMD_SAVE_DATABASE, NULL, NULL);` in `signals.c`). Hence it exists.

As a result, there is nothing stopping a developer from modifying the definition of `command_info_array[]` (e.g. add a line). The `sanity_check()` is the only way to catch the bug at compile time and preserve consistency.